contrib/storage-hive/core/src/main/codegen/templates/ObjectInspectorHelper.java

```diff
@@ -204,7 +204,11 @@ public static JBlock getDrillObject(JCodeModel m, ObjectInspector oi,
           <#elseif entry.hiveType == "TIMESTAMP">
             JVar tsVar = jc._else().decl(m.directClass(java.sql.Timestamp.class.getCanonicalName()), "ts",
               castedOI.invoke("getPrimitiveJavaObject").arg(returnValue));
-            jc._else().assign(returnValueHolder.ref("value"), tsVar.invoke("getTime"));
+            // Bringing relative timestamp value without timezone info to timestamp value in UTC, since Drill keeps date-time values in UTC
```

Comment:
Drill does not actually keep values in UTC: Drill keeps values in the server time zone. This is why Drill dates, when shipped to a client with a different time zone, get corrupted.